pkg/operator/configobservation/configobservercontroller/observe_config_controller.go

```diff
@@ -0,0 +1,50 @@
+package configobservercontroller
+
+import (
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	kubeinformers "k8s.io/client-go/informers"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/configobservation/builds"
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/configobservation/deployimages"
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/operator/configobservation/images"
+)
+
+type ConfigObserver struct {
+	*configobserver.ConfigObserver
+}
+
+// NewConfigObserver initializes a new configuration observer.
+func NewConfigObserver(
+	operatorClient v1helpers.OperatorClient,
+	configInformers configinformers.SharedInformerFactory,
+	kubeInformersForOperatorNamespace kubeinformers.SharedInformerFactory,
+	eventRecorder events.Recorder,
+) *ConfigObserver {
+	c := &ConfigObserver{
+		ConfigObserver: configobserver.NewConfigObserver(
+			operatorClient,
+			eventRecorder,
+			configobservation.Listers{
+				ImageConfigLister: configInformers.Config().V1().Images().Lister(),
+				ImageConfigSynced: configInformers.Config().V1().Images().Informer().HasSynced,
+				BuildConfigLister: configInformers.Config().V1().Builds().Lister(),
+				BuildConfigSynced: configInformers.Config().V1().Builds().Informer().HasSynced,
+				ConfigMapLister:   kubeInformersForOperatorNamespace.Core().V1().ConfigMaps().Lister(),
+				ConfigMapSynced:   kubeInformersForOperatorNamespace.Core().V1().ConfigMaps().Informer().HasSynced,
+			},
+			images.ObserveInternalRegistryHostname,
+			builds.ObserveBuildControllerConfig,
+			deployimages.ObserveControllerManagerImagesConfig,
+		),
+	}
+
+	kubeInformersForOperatorNamespace.Core().V1().ConfigMaps().Informer().AddEventHandler(c.EventHandler())
+	configInformers.Config().V1().Images().Informer().AddEventHandler(c.EventHandler())
```

Comment:
Need an event handler for `Builds`, too.

Reply:
well that is interesting, because the existing code didn't have one either, so i'm not sure how this was working.....

Reply:
sounds like we need an integration test. Given that we'll need to change cluster config to drive the test, not sure we have a clear path to build these tests at present.

Reply:
it can be done in our e2e suite.

Reply:
https://github.com/openshift/origin/pull/21864 starts us down that path. Each test adds 1-2 minutes to the total suite.

Reply:
I actually meant it can be added in the e2e suite on this repo, that's what i ended up doing.  (the test that sets the cluster config and ensures it propagates to the operand config).

Reply:
yes, but I think this gets us halfway there. IMO a full e2e should:

1. Modify the cluster config
2. Verify the operator processed the config change
3. Verify the config change rolls out to the operand
4. Verifies the expected behavior changed (ex: with `BuildDefaults`)

I'd love to test steps 3 and 4 in this repo, but atm origin has better test frameworks on this front.